todolist copy 3/main.js

```diff
@@ -0,0 +1,103 @@
+"use strict";
+
+const addTaskButton = document.getElementById('addTask');
+const lineTaskInput = document.getElementById('taskInput');
+const toDoList = document.getElementById('tasklist');
+const clearing = document.getElementById('clearAll');
+const checkBoxCover = document.getElementById('clearAll');
+
+
+let tasks = JSON.parse(localStorage.getItem("tasks")) ?? [];
+!localStorage.tasks ? tasks = [] : tasks = JSON.parse(localStorage.getItem('tasks'));
+
+function Task(description) {
+	this.description = description;
+	this.completed = false
+}
+
+let toDoElem = [];
+
+const updateHTML = () => {
+		toDoList.innerHTML = "";
+		if(tasks.length > 0){
+			tasks.forEach((item, index) => {
+				let taskLine = document.createElement('li');
+	taskLine.classList.add("task-item");
+
+	let deleteButton = document.createElement('button');
+	deleteButton.classList.add("delete");
+	deleteButton.innerHTML = '<input type="image" class="trash" src="pics/trash.svg" name="delete" alt="delete">';
+	deleteButton.onclick = function() {deleteTask(index)};
+
+	let taskLabel = document.createElement('label');
+	taskLabel.classList.add("tasks");
+
+	let checkBoxComplete = document.createElement('input');
+	checkBoxComplete.classList.add("complete");
+	checkBoxComplete.type = 'checkbox';
+	checkBoxComplete.onclick = function() {completeTask(index)};
+
+	let taskName = document.createElement('span');
+	taskName.classList.add("task-name");
+	taskName.innerHTML = item.description;
+
+	let checkMark = document.createElement('span');
+	checkMark.classList.add("checkmark");
+
+	taskLabel.appendChild(checkBoxComplete);
+	taskLabel.appendChild(taskName);
+	taskLabel.appendChild(checkMark);
+
+	taskLine.appendChild(deleteButton);
+	taskLine.appendChild(taskLabel);
+
+	toDoList.appendChild(taskLine);
+	
+			})
+			toDoElem = document.querySelectorAll('.task-item'); 
+		}
+}
+updateHTML();
+	const updateLocal = () => {
+	localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+const completeTask = index => {
+	tasks[index].completed = !tasks[index].completed; 
+	if(tasks[index].completed) {

```

Comment:
our source of truth is data, it means that we shouldn't do manual work of updating classes, we should do it while we are updating our HTML